apps-rendering/src/components/TableOfContents/index.tsx

```diff
@@ -0,0 +1,188 @@
+import type { SerializedStyles } from '@emotion/react';
+import { css } from '@emotion/react';
+import { text } from '@guardian/common-rendering/src/editorialPalette/text';
+import type { ArticleFormat } from '@guardian/libs';
+import {
+	line,
+	neutral,
+	remSpace,
+	textSans,
+} from '@guardian/source-foundations';
+import {
+	SvgChevronDownSingle,
+	SvgChevronUpSingle,
+} from '@guardian/source-react-components';
+import Anchor from 'components/Anchor';
+import ListItem from 'components/ListItem';
+import OrderedList from 'components/OrderedList';
+import type { Outline } from 'outline';
+import type { FC, ReactElement } from 'react';
+import { darkModeCss } from 'styles';
+
+interface Props {
+	format: ArticleFormat;
+	outline: Outline;
+}
+
+interface TextElementProps {
+	node: Node;
+	key: number;
+}
+
+const anchorStyles = (format: ArticleFormat): SerializedStyles => css`
+	color: ${text.paragraph(format)};
+	border-bottom: none;
+	:hover {
+		border-bottom: 0.0625rem solid ${neutral[86]};
+	}
+`;
+
+const listStyles: SerializedStyles = css`
+	> li::before {
+		content: none;
+	}
+
+	margin: 0;
+
+	> li {
+		padding-left: ${remSpace[3]};
+	}
+`;
+
+const listItemStyles: SerializedStyles = css`
+	${textSans.xsmall({ fontWeight: 'bold', lineHeight: 'regular' })}
+	border-bottom: ${line.primary} 1px solid;
+	padding-top: ${remSpace[2]};
+`;
+
+const detailsStyles: SerializedStyles = css`
+	margin-bottom: ${remSpace[6]};
+	&:not([open]) .is-on,
+	&[open] .is-off {
+		display: none;
+	}
+	summary::-webkit-details-marker {
+		display: none;
+	}
+`;
+
+const summaryStyles: SerializedStyles = css`
+	cursor: pointer;
+	position: relative;
+	list-style: none;
+	align-items: center;
+	padding-left: ${remSpace[3]};
+	padding-top: 0.44rem;
+	padding-bottom: 0.375rem;
+	border-bottom: ${line.primary} 1px solid;
+	border-top: ${line.primary} 1px solid;
```

Comment:
It's possible that it doesn't matter, but I think the CSS spec indicates the following order for these properties: `width style color`, which would give:

```suggestion
	border-bottom: 1px solid ${line.primary};
	border-top: 1px solid ${line.primary};
```

There's a bit more from the spec on [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/border#formal_syntax).

Reply:
Thanks @georgeblahblah Done